src/main/scala/logging/LogCollection.scala

```diff
@@ -25,39 +21,24 @@ object LogCollection {
 
 
 case class LogCollection(activeLoggers:Seq[Logger]) {
-  val DEFAULT_SLEEP_DELAY = 1000 //milliseconds
-  val logQueue = new ConcurrentLinkedQueue[Option[LogMessage]]
-
-  class logProcessor(logQueue:ConcurrentLinkedQueue[Option[LogMessage]]) extends Runnable {
-    override def run(): Unit = {
-      var shouldExit:Boolean = false
-      do {
-        logQueue.poll() match {
-          case null=>Thread.sleep(1000)
-          case Some(msg)=>
-            activeLoggers.foreach(x=>x.relayMessage(msg))
-          case None=>
-            shouldExit=true
-        }
-      } while(!shouldExit)
-    }
-  }
 
   def activeLoggerCount:Integer = activeLoggers.size
 
-  //def relayMessage(msg:String,currentMethod:CDSMethod,severity:String) = activeLoggers.foreach(x=>x.relayMessage(msg,currentMethod,severity))
-  def relayMessage(msg:String, sender:CDSMethod, severity: String) = logQueue.add(Some(LogMessage(msg,severity,sender)))
-  def relayMessage(m:LogMessage) = logQueue.add(Some(m))
+  def relayMessage(msg:LogMessage):Seq[Future[Unit]] = activeLoggers.map(x=>x.relayMessage(msg))
+
```

Comment:
you can use underscore here if you to avoid having to create a throwaway var name:
```scala
activeLoggers.map(_.relayMessage(msg))
```